File: common/src/main/java/com/codeloom/common/language/SubmissionLanguageValidator.java
package com.codeloom.common.language;

import jakarta.validation.ConstraintValidator;
import jakarta.validation.ConstraintValidatorContext;

public class SubmissionLanguageValidator implements ConstraintValidator<ValidLanguage, String> {
    @Override
    public boolean isValid(String value, ConstraintValidatorContext context) {
        if (value == null) return false;
        try {
            LanguageSpec.fromLanguage(value);
            return true;
        } catch (InvalidLanguageException ignored) {
            return false;
        }
    }
}

File: common/src/main/kotlin/com/codeloom/common/language/SubmissionLanguageValidator.kt
package com.codeloom.common.language

import jakarta.validation.ConstraintValidator
import jakarta.validation.ConstraintValidatorContext

class SubmissionLanguageValidator : ConstraintValidator<ValidLanguage, String> {
    override fun isValid(
        value: String?,
        context: ConstraintValidatorContext?
    ): Boolean {
        if (value == null) {
            return false;
        }

        try {
            LanguageSpec.fromLanguage(value)
        } catch (_: InvalidLanguageException) {
            return false;
        }

        return true
    }
}